src/remote.go

```diff
@@ -152,39 +151,69 @@ func checkServer() {
 		mc.Conn.ServerColor = "red"
 		mc.Conn.ServerStatus = "FAIL"
 	} else {
-		if resp.StatusCode == 200 {
-			mc.Conn.ServerColor = "green"
-			mc.Conn.ServerStatus = "OK"
-		} else {
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			failPrint("Error reading Status body.")
 			mc.Conn.ServerColor = "red"
-			mc.Conn.ServerStatus = "ERROR"
+			mc.Conn.ServerStatus = "FAIL"
+		} else {
+			handleStatus(string(body))
+			if resp.StatusCode == 200 {
+				mc.Conn.ServerColor = "green"
+				mc.Conn.ServerStatus = "OK"
+			} else {
+				mc.Conn.ServerColor = "red"
+				mc.Conn.ServerStatus = "ERROR"
+			}
 		}
 	}
 
 	// Overall
 	if mc.Conn.NetStatus == "FAIL" && mc.Conn.ServerStatus == "OK" {
+		timeStart = time.Now()
 		mc.Conn.OverallColor = "goldenrod"
 		mc.Conn.OverallStatus = "Server connection good but no Internet. Assuming you're on an isolated LAN."
 		mc.Connection = true
 	} else if mc.Conn.ServerStatus == "FAIL" {
+		timeStart = time.Now()
 		mc.Conn.OverallColor = "red"
 		mc.Conn.OverallStatus = "Failure! Can't access remote scoring server."
 		failPrint("Can't access remote scoring server!")
 		sendNotification("Score upload failure! Unable to access remote server.")
 		mc.Connection = false
 	} else if mc.Conn.ServerStatus == "ERROR" {
+		timeWithoutId = time.Now().Sub(timeStart)
+		if !mc.Config.NoDestroy && timeWithoutId > withoutIdThreshold {
+			failPrint("Destroying the image! Too long without inputting valid ID.")
+			// destroyImage()
+		}
 		mc.Conn.OverallColor = "red"
-		mc.Conn.OverallStatus = "Score upload failure. Can't send scores to remote server."
-		failPrint("Remote server returned an error for its status!")
-		sendNotification("Score upload failure! Remote server returned an error.")
+		mc.Conn.OverallStatus = "Scoring engine rejected your TeamID!"
+		failPrint("Remote server returned an error for its status! Your ID is probably wrong.")
+		sendNotification("Status check failed, TeamID incorrect!")
 		mc.Connection = false
 	} else {
+		timeStart = time.Now()
 		mc.Conn.OverallColor = "green"
 		mc.Conn.OverallStatus = "OK"
 		mc.Connection = true
 	}
 }
 
+func handleStatus(status string) {
+	switch status {
+	// Please no comments on how I'm handling parsing JSON
```

Comment:
:( can we actually unmarshal the JSON here in case the response is weird? i can do it rq if u dont want to

Reply:
wdym bro its top tier json parsing